blockmanager.go

```diff
@@ -873,6 +913,36 @@ func (b *blockManager) handleInvMsg(imsg *invMsg) {
 		}
 	}
 
+	// If this inv contains a block annoucement, and this isn't coming from
+	// our current sync peer or we're current, then update the last
+	// announced block for this peer. We'll use this information later to
+	// update the heights of peers based on blocks we've accepted that they
+	// previously announced.
+	if lastBlock != -1 && (imsg.peer != b.syncPeer || b.current()) {
+		imsg.peer.UpdateLastAnnouncedBlock(&invVects[lastBlock].Hash)
+	}
+
+	// Ignore invs from peers that aren't the sync if we are not current.
+	// Helps prevent fetching a mass of orphans.
+	if imsg.peer != b.syncPeer && !b.current() {
+		return
+	}
+
+	// If our chain is current and a peer announces a block we already
+	// know of, then update their current block height.
+	if lastBlock != -1 && b.current() {
+		exsits, err := b.server.db.ExistsSha(&invVects[lastBlock].Hash)
+		if err == nil && exsits {
+			blkHeight, err := b.server.db.FetchBlockHeightBySha(&invVects[lastBlock].Hash)
+			if err != nil || blkHeight == 0 {
```

Comment:
Why the check for 0?  If it is 0 for some reason (without err=nil) it will print something ugly like "<nil>" when it logs this below.